src/lib/storage/chunk.hpp

```diff
@@ -183,6 +191,12 @@ class Chunk : private Noncopyable {
     return index;
   }
 
+  template <typename Index>
+  std::shared_ptr<BaseIndex> create_index(const std::vector<ColumnID>& column_ids) {
+    const auto columns = get_columns_for_ids(column_ids);
+    return create_index<Index>(columns);
```

Comment:
When I call `Chunk::create_index`, I would expect the chunk to keep the created index, not return it. (I know that it was there before.)